### PR TITLE
Refactor kubectl label to untie CLI flags from runtime options

### PR DESCRIPTION
**Pull request by jaehnri**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

This PR unties CLI flags from runtime options in the `kubectl label` command. 

#### Which issue(s) this PR fixes:

Ref https://github.com/kubernetes/kubectl/issues/1046

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by dims** [comment]
If you still need this PR then please rebase, if not, please close the PR

**Comment by jaehnri** [comment]
Hi, @dims!
I'll rebase it soon. I need a review here, do you know if I should call someone specifically?

File: staging/src/k8s.io/kubectl/pkg/cmd/label/label.go [hunk 1]
```diff
@@ -140,76 +186,92 @@ func NewCmdLabel(f cmdutil.Factory, ioStreams genericclioptions.IOStreams) *cobr
...
-	o.builder = f.NewBuilder()
-	o.unstructuredClientForMapping = f.UnstructuredClientForMapping
+	builder := f.Factory.NewBuilder()
+	unstructuredClientForMapping := f.Factory.UnstructuredClientForMapping
 
-	return nil
+	return &LabelOptions{
+		FilenameOptions: f.FilenameOptions,
+
+		PrintFlags: f.PrintFlags,
```
Comment (ardaguclu) [inline staging/src/k8s.io/kubectl/pkg/cmd/label/label.go:247]: We already have `ToPrinter`, why we are also storing `PrintFlags` in options?

File: staging/src/k8s.io/kubectl/pkg/cmd/label/label_test.go [hunk 2]
```diff
@@ -696,18 +699,20 @@ pod/foo not labeled
...
 			}
-			options := NewLabelOptions(iostreams)
+
+			flags := NewLabelFlags(tf, iostreams)
+			options, err := flags.ToOptions(cmd, tc.args)
 			if tc.overwrite {
 				options.overwrite = true
 			}
-			if err := options.Complete(tf, cmd, tc.args); err != nil {
+			if err != nil {
```
Comment (ardaguclu) [inline staging/src/k8s.io/kubectl/pkg/cmd/label/label_test.go:708]: if options is nil, above assignment will panick `options.ovewrite = true`. Maybe we can change the order and first check `err != nil` before assignment?

**Review by ardaguclu — COMMENTED** [review]
Thanks @jaehnri for the PR.

It looks good to me, left a few comments.

File: staging/src/k8s.io/kubectl/pkg/cmd/label/label_test.go (hunk 2 above)
Comment (jaehnri) [reply staging/src/k8s.io/kubectl/pkg/cmd/label/label_test.go:705]: Well noticed. Thanks!

File: staging/src/k8s.io/kubectl/pkg/cmd/label/label.go (hunk 1 above)
Comment (jaehnri) [reply staging/src/k8s.io/kubectl/pkg/cmd/label/label.go]: Hmm, your review made me realize the difference between Flags and Options. Did it, thanks!

File: staging/src/k8s.io/kubectl/pkg/cmd/label/label.go [hunk 3]
```diff
@@ -50,11 +50,69 @@ const (
...
+// the logic itself easy to unit test
+type LabelFlags struct {
+	Factory cmdutil.Factory
+
+	resource.FilenameOptions
+
+	RecordFlags *genericclioptions.RecordFlags
+	PrintFlags  *genericclioptions.PrintFlags
+
+	Recorder genericclioptions.Recorder
```
[the remark this answers is not in the document]
Comment (jaehnri) [reply staging/src/k8s.io/kubectl/pkg/cmd/label/label.go]: Yup, this way we also got rid of the `genericclioptions.NoopRecorder{}` in Flags.

**Pull request closed without merging**